packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -415,9 +417,35 @@ export class TempCollabSpaceRuntime
 		}
 		this.matrix.switchSetCellPolicy();
 
+		// We need to ensure that we are tracking all the cells and when removing cells, we need to
+		// remove the tracking information as well as update the indexes in the map.
+		const removeCellsFromMap = (start: number, count: number, map: Map<string, any>) => {
+			if (count > 0) {
+				// Convert the map to an array of key-value pairs
+				const entries = Array.from(map.entries());
+
+				// Remove the specified range of items from the array
+				// Note the index from the reverse mapping is off by 1 as the matrix has a row and col tracking IDs
+				// and the map indexes are 0 based
+				entries.splice(start - 2, count);
```

Comment:
I'm not sure how to think about order here. when you convert a map into array, what dictates the order of the entries in array? The values will be inserted randomly into this map (as user adds and deletes rows). So it sounds like the order here is simply random, and I do not think we can assume anything.

Before fixing anything, can you please add a unit test that hits that problem? Such that we have coverage for future?

Reply:
Or is logic in insertCellIntoReverseMap() tries to enforce some order? If it does, are you sure that map.entries() provides us some guarantees? I do not see anything listed on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/entries, though I have not looked at full spec

Reply:
It should always keep the values in order on the Map. I will add more testing  and also look at the Tablero implementation as I mentioned.